### PR TITLE
Deprecate Get_Program

### PR DESCRIPTION
**Pull request by eric-wieser**
This function was basically useless. The behavior before was that:

* `Get_Program(); Print_Function()` is the same as `Print_Function()`
* `Get_Program(False); Print_Function()` is the same as `Print_Function()`
* `Get_Program(True); Print_Function()` does nothing at all

It's simpler to just not call `Print_Function` if you don't want to print the function.

### CODE REVIEW

**Comment by utensil** [comment]
get program should b preserved with its on-off switch, print function should be dreprecatd instead due t poor naming.

**Comment by eric-wieser** [comment]
If we deprecated `Print_function`, then the on-off switch of `Get_Program` would do nothing at all. Remember, the code in master is:
```python
# used only in the two functions below
off_mode = False

def Get_Program(off=False):
    global off_mode
    off_mode = True

def Print_Function():
    if off_mode
       return  # don't print anything
    _actually_print_function()
```
This pr takes the view that there's no reason for `off_mode` to exist, and if users do not want to print the function source, they should not be calling a function called `Print_Function`.

**Comment by utensil** [comment]
How about deprecate them in one go and replace with a better API? Never liked the setup, naming and casing/underscore.

**Comment by eric-wieser** [comment]
> How about deprecate them in one go and replace with a better API? Never liked the setup, naming and casing/underscore.

Not sure there's any huge advantage to deprecating them both at once. The whole file has some questionably-named functions, and it's probably best to rename them all at once, in a future release.

**Comment by eric-wieser** [comment]
> replace with a better API?

This is a little tricky, because `Print_Function` actually is coupled with `xpdf`, as the former emits special `##` markers that are interpreted as escape sequences by the latter. It's generally a bit of a mess, and if we're going to introduce a new API, I'd rather we did it after removing that mess so that we don't end up introducing a second new API later.

File: galgebra/printer.py [hunk 1]
```diff
@@ -1089,9 +1091,16 @@ def LatexFormat(Fmode=True, Dmode=True, ipy=False):
...
+        "galgebra.printer.Get_Program is deprecated, and exists solely to "
+        "toggle whether galgebra.printer.Print_Function does anything. If you "
+        "want to turn off program printing, then just don't call Print_Function!",
+        DeprecationWarning, stacklevel=2)
 
 
 def Print_Function():
+    """ Print out the source of the current function """
     if off_mode:
         return
```
Comment (utensil) [inline galgebra/printer.py:1105]: Why not remove this as well?

Comment (eric-wieser) [reply galgebra/printer.py:1105]: Because that would be a breaking change - once we remove `Get_Program`, we can remove this.

Comment (eric-wieser) [reply galgebra/printer.py:1016]: I could stick a deprecation warning here too, but that would probably be overkill

Comment (utensil) [reply galgebra/printer.py:1016]: Oh

Comment (utensil) [reply galgebra/printer.py:1016]: Let's go ahead and merge as-is. Sorry I wasn't thinking clearly. Now it doesn't seem to be a problem.

**Pull request merged**